xbmc/video/VideoDatabase.cpp

```diff
@@ -3950,44 +3939,73 @@ void CVideoDatabase::GetSameVideoItems(CFileItem& item, CFileItemList& items)
   {
     // get items with same unique ids
     m_pDS->query(
-        PrepareSQL("SELECT media_id "
+        PrepareSQL("SELECT DISTINCT media_id "
                    "FROM uniqueid "
                    "WHERE (value, type) IN "
                    "  (SELECT value, type FROM uniqueid WHERE media_id = %i AND media_type = '%s')",
```

Comment:
sorry, haven't noticed before, but this query seems fishy or I just don't understand what you are trying to do here. Why do you need a subselect on the table you query in your main query? Wouldn't this query return the same result?
```
"SELECT DISTINCT media_id "
"FROM uniqueid "
"WHERE media_id = %i AND media_type = '%s')"
```

Reply:
There is comment for this above the query. It's used to get all media ids which have the same unique ids as the specified media id. What your query above just returns the media id you specified. You can check the uniqueid table to see how it works.

Reply:
You can't have an item with the same unique id, else it wouldn't be unique!  What the query fetches is actually a distinct value for the media id.  However as you are already specifying a media_id it can only ever fetch that specified value.  EG, if I have three movie versions that have a media_id (in the uniqueid table) of say 55, then a query of 
`SELECT distinct media_id FROM uniqueid WHERE (value, type) IN (SELECT value, type FROM uniqueid WHERE media_id = 55 AND media_type = 'movie')`  will just return one row of 55, irrespective of the fact that there are 3 versions.  The uniqueid bit doesn't seem to come in to it.

@da-anda , do you agree ?

Reply:
The comment is technically correct but doesn't help much. Maybe something like "Retrieve items that share a metadata identifier: alternative versions imported before versions feature, or a version merge was rejected by the user"

Yes @da-anda the `WHERE (value, type) IN (SELECT value, type...` eliminates the rows that don't match both fields.

Reply:
Please provide a more descriptive comment for our future selves, see couple comments above.